### DISCUSSION TITLE
Bug: Invalid interpolation format for "environment" option

### DISCUSSION THREAD

**Discussion by jorroll** [community/community#40493] [Codespaces]
### Select Topic Area

Bug

### Body

Codespacess is failing to build my devcontainer with the following error.

```
Invalid interpolation format for "environment" option in service "graphql-engine": "{"type":"HS256", "key": "secretsecretsecretsecretsecretsecret", "claims_namespace_path": "$.claims.hasura"}"
```

The problem appears to be that the devcontainer's `docker-compose.yaml` file contains an environment variable which is a JSON formatted string (i.e. `'{"type":"HS256", "key": "secretsecretsecretsecretsecretsecret", "claims_namespace_path": "$.claims.hasura"}'`). 

It appears as though codespaces is grabbing that envirnment variable and then reinserting it somewhere else in the form `"${JORROLS_ENV_VARIABLE}"`. The problem being that this expands to `"{"type":"HS256", "key": "secretsecretsecretsecretsecretsecret", "claims_namespace_path": "$.claims.hasura"}"` which is invalid. 

In this case, it should be inserted like `'${JORROLS_ENV_VARIABLE}'` which will expand to `'{"type":"HS256", "key": "secretsecretsecretsecretsecretsecret", "claims_namespace_path": "$.claims.hasura"}'`.

**Comment by ossanna16** [comment]
Hi there @jorroll and welcome to our community! Thank you for reporting this!
